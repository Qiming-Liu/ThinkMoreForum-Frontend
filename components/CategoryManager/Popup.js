import React from 'react';
import { Dialog, DialogContent, DialogTitle, Typography } from '@mui/material';
import styled from 'styled-components';
import CloseIcon from '@mui/icons-material/Close';
import Controls from './controls/Controls';

const StyDialog = styled(Dialog)`
  padding: 16px;
  position: absolute;
  top: 40px;
`;

const DialogWrapper = styled(DialogTitle)`
  padding-right: 0px;
`;

const Popup = (props) => {
  const { children, openPopup, setOpenPopup } = props;
  return (
    <StyDialog open={openPopup} maxWidth="md">
      <DialogWrapper>
        <div style={{ display: 'flex' }}>
          <Typography variant="h6" component="div" style={{ flexGrow: 1 }}>
            llll
          </Typography>
          <Controls.ActionButton
            color="secondary"
            onClick={() => {
              setOpenPopup(false);
            }}
          >
            <CloseIcon />
          </Controls.ActionButton>
        </div>
      </DialogWrapper>
      <DialogContent dividers>{children}</DialogContent>
    </StyDialog>
  );
};

export default Popup;
