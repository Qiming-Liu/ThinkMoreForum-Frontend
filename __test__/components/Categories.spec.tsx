import React from 'react';
import { render } from '@testing-library/react';
import Popup from '../../components/CategoryManager/Popup';

describe('<Popup />', () => {
  it('should render nothing when setOpenPopup is false', () => {
    const { queryByText } = render(
      <Popup openPopup={false} setOpenPopup={() => {}} />,
    );
    expect(queryByText('New category')).toBeNull();
  });
  it('should render CategoryForm when setOpenPopup is true', () => {
    const { container } = render(
      <Popup openPopup setOpenPopup={() => {}}>
        This is category form
      </Popup>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
