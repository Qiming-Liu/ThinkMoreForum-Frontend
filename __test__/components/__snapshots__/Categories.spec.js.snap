// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> should render CategoryForm when setOpenPopup is true 1`] = `null`;
